hoomd/BoxDim.h

```diff
@@ -496,6 +496,139 @@ struct
         w.z = v.z;
         }
 
+    //! Wrap vectors back into the box
+    /*! \param pos Vector to wrap, updated to the minimum image obeying the periodic settings
+        \param vel Vector to wrap according to deformation rates, obeying the periodic settings
+        \param img Image of the vector, updated to reflect the new image
+        \param flags Vector of flags to force wrapping along certain directions
+        \post \a img and \a v are updated appropriately
+        \note \a v must not extend more than 1 image beyond the box
+    */
+   
+    HOSTDEVICE void wrap_pos_vel(Scalar3& pos, Scalar3& vel, int3& img, char3 flags = make_char3(0, 0, 0)) const
+        {
+        Scalar3 L = getL();
+
+        // allow for a shifted box with periodic boundary conditions
+        Scalar3 origin = (m_hi + m_lo) / Scalar(2.0);
+
+        // tilt factors for nonperiodic boxes are always calculated w.r.t. to the global box with
+        // origin (0,0,0)
+        if (!m_periodic.y)
+            {
+            origin.y = Scalar(0.0);
+            }
+        if (!m_periodic.z)
+            {
+            origin.z = Scalar(0.0);
+            }
+
+        if (m_periodic.x)
+            {
+            Scalar tilt_x = (m_xz - m_xy * m_yz) * (pos.z - origin.z) + m_xy * (pos.y - origin.y);
+            Scalar Lx_shear;
```

Comment:
Shouldn't you be able to just use the member variables you added (or combinations of them with other member variables)?